File: tools/bench/polyglot/python/fib_iter.py
# Fixed-width wrap (like i64) so Python does not grow into bigints.
# Matches wrapping add on Ori/Rust 64-bit integers.
MASK = (1 << 64) - 1
n = 20_000_000
a, b = 0, 1
i = 0
while i < n:
    a, b = b, (a + b) & MASK
    i += 1
# signed i64 print for result parity with wrapping languages
if a >= (1 << 63):
    a -= 1 << 64
print(a)

File: tools/bench/polyglot/ruby/fib_iter.rb
# 64-bit wrap (like i64) — avoid Ruby bigint blow-up dominating the kernel.
MASK = (1 << 64) - 1
n = 20_000_000
a = 0
b = 1
i = 0
while i < n
  t = (a + b) & MASK
  a = b
  b = t
  i = i + 1
end
a -= (1 << 64) if a >= (1 << 63)
puts a
